api/login.ts

```diff
@@ -21,10 +21,20 @@ export default async function handler(req: VercelRequest, res: VercelResponse) {
 
     try {
       const message = new SiweMessage(data);
+
+      // TODO: replace by configured domain
+      // TODO: Include this in .env?
+      const domain = 'localhost:3000';
+      if (message.domain !== domain) {
+        return errorResponse(res, {
+          message: `domain in message doesnt match ${domain}`,
+          httpStatus: 401,
+        });
+      }
```

Comment:
This verification is included in `message.verify`.
To return a custom error message specifically for this check, you can use `try/catch` and check `e.error.type`.

```js
try {
    message.verify({ signature, domain });
} catch (e) {
    if (e.error.type == SiweErrorType.DOMAIN_MISMATCH){
        return errorResponse(res, {
          message: `domain in message doesnt match ${domain}`,
          httpStatus: 401,
        });
    } else {
        // Treat other SIWE exceptions and/or return generic error
    }
}
```

Reply:
oh gotcha, thanks! will include that in the PR that I'll make to the main repo